### DISCUSSION TITLE
combining coverage reports

### DISCUSSION THREAD

**Discussion by stephencranedesign** [vitest-dev/vitest#5496] [Q&A]
what's the best way to combine coverage reports run with a different tool? we're migrating from mocha with nyc to vitest. the approach our team is currently using is a set of manual excludes automatically generated in both coverage config tools based on a .spec. vs .test. file. I'd like to be able to run both test suites tho and then combine the coverage info from both runs and generate a single report. Is that possible?

**Comment by AriPerkkio** [comment]
It's possible. Make both test runners output the JSON coverage report and use `nyc` to merge those. See https://github.com/istanbuljs/nyc#combining-reports-from-multiple-runs.

Note that `nyc` is compatible with Vitest's `coverage: { provider: 'istanbul' }` only.

Related discussions: 
- https://github.com/vitest-dev/vitest/discussions/4755
- https://github.com/vitest-dev/vitest/discussions/3744

**Comment by stephencranedesign** [comment]
@AriPerkkio 

yeah I saw those and spent some time trying to make them work without success. here's a sample repo I put together to replicate my project demonstrating that coverage is failing when I try to combine reports on shared code between the 2 different test runners: https://github.com/stephencranedesign/vitest-coverage-question.

it seems like combining coverage is possible in theory but not really in practice.

**Reply by AriPerkkio** [reply]
Looks like the coverage reports produced by Mocha and Vitest are a bit different. I would guess that this is caused by different source maps. `nyc merge` seems really strict and isn't able to merge these reports then.
